### DISCUSSION TITLE
definePropsRefs return type

### DISCUSSION THREAD

**Discussion by extrem7** [vue-macros/vue-macros#545] [Q&A] [answered]
Docs describe return type as `Ref<>` but actually it's `ComputedRef<DeepReadonly<>>` that produces a lot of TS errors when migrating project from `toRefs` to `definePropsRefs`: when prop ref is passed into composable function its type should be `Ref<readonly` or so.

![image](https://github.com/vue-macros/vue-macros/assets/24792876/4c8e1694-1f7a-4682-801c-18124be8c86a)

**Comment by sxzz** [comment, marked as answer]
The returning type of `definePropsRefs` is stricter than `toRefs`. Because `toRefs` doesn't know the context in handling props. For `definePropsRefs` we know the props should be readonly and never be assigned.

**Reply by extrem7** [reply]
Should it be described in docs?

**Reply by sxzz** [reply]
I think it's worth to mention this point. PR welcome.
